api/services.py

```diff
@@ -1,38 +1,97 @@
+from api.routes import get_price_levels, get_db, get_emails_by_zone
 from datetime import datetime
+import time
+import smtplib
+import logging
+from email.mime.text import MIMEText
+
 
 def when_to_notify(zone: str, db: Session = Depends(get_db)):
-    # this gets the high price of the last month
-    price_levels = get_price_levels(zone=zone, db=db)
-    high_price_last_month = price_levels["priceLevels"]["high"]
-    current_time = datetime.utcnow()
-
-    # this gets the current price
-    query_current_price = text("""
-        SELECT price_sek 
-        FROM price_data 
-        WHERE zone = :zone 
-        AND time_start >= :current_time
-        ORDER BY time_start ASC 
+    try:
+        # this gets the high price of the last month
+        price_levels = get_price_levels(zone=zone, db=db)
+        high_price_last_month = price_levels["priceLevels"]["high"]
+        
+        current_time_unix = int(time.time())
+        # this gets the prices for today
+        query_daily_prices = text("""
+            SELECT price_sek, time_start
+            FROM price_data 
+            WHERE zone = :zone 
+            AND time_start >= :current_time_unix
+            ORDER BY time_start ASC 
     """)
-    current_price_result = db.execute(query_current_price, {"zone": zone}).fetchone()
+        daily_prices_result = db.execute(query_daily_prices, {"zone": zone, "current_date": current_time_unix}).fetchall()
+        if daily_prices_result:
+            # Extract prices and their corresponding timestamps
+            daily_prices = [(float(price[0]), price[1]) for price in daily_prices_result]  # Assuming price[1] is the timestamp
+            high_price_today, high_price_time = max(daily_prices, key=lambda x: x[0])  # Get the highest price and its time
+            logger.info(f"High price in zone {zone} for today: {high_price_today} SEK at {high_price_time}")
+    
+            # Check if today's high price exceeds last month's high price
+            if high_price_today > high_price_last_month:
+                # Get emails of subscribers in the specified zone
+                emails = get_emails_by_zone(zone=zone, db=db)
+                
+                # Prepare email details
+                subject = f"High Electricity Price Alert for {zone}"
+                body = "The electricity price has exceeded last month's high price."
+    
+                # Schedule email sending for the time of the peak price
+                schedule_email_once(emails, subject, body, high_price_today, high_price_time)
+    
+        else:
+            logger.warning(f"No price data found for zone {zone} on {current_date}.")
+    except Exception as e:
+        logger.error(f"Error en when_to_notify: {str(e)}")
+
+# Function to send the email
+def send_email(to_address: str, subject: str, body: str, price: float):
+    smtp_server = "smtp.gmail.com"
+    smtp_port = 587
+    sender_email = "team3.projectagile@gmail.com" 
+    sender_password = "Team3project"  
```

Comment:
This is private information and we should import it from our git-ignored config file